extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js

```diff
@@ -673,16 +714,24 @@ export class AmpLightboxViewer extends AMP.BaseElement {
   resizeImageViewerDimensions_() {
     const imgViewer = this.elementsMetadata_[this.currentElemId_].imageViewer;
 
-    // Unregister the previous onResize handler to rescale the ImageViewer
     if (this.unlistenResize_) {
       this.unlistenResize_();
     }
 
-    // Register a new onResize handler
+    // Register an onResize handler to resize the image viewer
+    const platform = Services.platformFor(this.win);
+    const boundOnResize = this.onResize_.bind(this);
+    const debouncedOnResize = debounce(this.win, boundOnResize, 500);
     this.unlistenResize_ = this.getViewport().onResize(() => {
```

Comment:
Would is make sense to encapsulate this hack into the onResize method here?

Reply:
Ping on this comment.

Reply:
As in this? 

```
registerOnResizeHandler_() {
    // Register an onResize handler to resize the image viewer
    this.unlistenResize_ = this.getViewport().onResize(() => {
        const platform = Services.platformFor(this.win);
        const onResize = this.resizeCurrentImageViewer_.bind(this);

        // Special case for iOS browsers due to Webkit bug #170595
        // https://bugs.webkit.org/show_bug.cgi?id=170595
        // Delay the onResize by 500 ms to ensure correct height and width
        const debouncedOnResize = debounce(this.win, onResize, 500);

        if (platform.isIos() && platform.isSafari()) {
            debouncedOnResize();
        } else {
            onResize();
        }
    });
  }
```

It would, except I moved the `orientationchange` code to live in the same block, so it is sharing the initialization code for `debouncedOnResize` and `platform`. Or maybe I'm not correctly understanding your comment?

Reply:
I think by encapsulate @cramforce meant to move it into the `viewport-impl` service

Reply:
This bug only manifest itself if one is measuring viewport width/height after a resize event. There are usages of `onResize` where there is no measurement of viewport dimensions and delaying these handlers by 500ms may not be great. (e.g. `amp-animation` restarts on resize).

Reply:
Cool. Lets handle it by adding a comment to `viewport.onResize` to help future users.

Reply:
OH. Sorry. Misunderstood this comment. Oops.

Reply:
Added a comment in `viewport-impl.js`'s `onResize` function.

Reply:
Apologies for the earlier confusion @cramforce. Feel free to lmk if anything else should be changed.